snuba/web/db_query.py

```diff
@@ -243,15 +245,33 @@ def execute_query_with_rate_limits(
     stats: MutableMapping[str, Any],
     query_settings: MutableMapping[str, Any],
     robust: bool,
+    applied_quota: Optional[RateLimitStats],
 ) -> Result:
     # XXX: We should consider moving this that it applies to the logical query,
     # not the physical query.
     with RateLimitAggregator(
         request_settings.get_rate_limit_params()
     ) as rate_limit_stats_container:
-        stats.update(rate_limit_stats_container.to_dict())
         timer.mark("rate_limit")
 
+        if applied_quota is not None:
+            project_rate_limit_stats: Optional[RateLimitStats] = applied_quota
+            rate_limit_stats_container.add_stats(QUOTA_RATE_LIMIT_NAME, applied_quota)
+        else:
+            project_rate_limit_stats = rate_limit_stats_container.get_stats(
+                PROJECT_RATE_LIMIT_NAME
+            )
+        stats.update(rate_limit_stats_container.to_dict())
+
+        if applied_quota is not None and project_rate_limit_stats is not None:
```

Comment:
I don't think your check here is correct. `project_rate_limit_stats` is set to be the same value as `applied_quota` if `applied_quota` is not None.